wls-broadcast-service/src/test/java/de/muenchen/oss/wahllokalsystem/broadcastservice/security/BroadcastSecurityTest.java

```diff
@@ -0,0 +1,181 @@
+import de.muenchen.oss.wahllokalsystem.broadcastservice.TestConstants;
+import de.muenchen.oss.wahllokalsystem.broadcastservice.rest.BroadcastMessageDTO;
+import de.muenchen.oss.wahllokalsystem.broadcastservice.service.BroadcastService;
+
+import de.muenchen.oss.wahllokalsystem.broadcastservice.utils.BroadcastSecurityUtils;
+import java.util.Arrays;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest(
+        classes = { MicroServiceApplication.class },
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles(profiles = { SPRING_TEST_PROFILE })
+@Slf4j
+public class BroadcastSecurityTest {
+
+    @Autowired
+    BroadcastService broadcastService;
+
+    @BeforeEach
+    void setUp() {
+        Assertions.assertThat(broadcastService).isNotNull();
+        SecurityContextHolder.clearContext();
+    }
+
+    @Test
+    void broadcastAccessDeniedTest() {
+        SecurityContextHolder.getContext().setAuthentication(
+                new UsernamePasswordAuthenticationToken(TestConstants.TESTUSER, TestConstants.TESTPASSWORD, AuthorityUtils.createAuthorityList("ROLE_DUMMY")));
+
+        Assertions.assertThatExceptionOfType(AccessDeniedException.class)
+                .isThrownBy(() -> broadcastService.broadcast(null)).withMessageStartingWith("Access Denied");
+    }
+
+    @Test
+    void getMessageAccessDeniedTest() {
+        SecurityContextHolder.getContext().setAuthentication(
+                new UsernamePasswordAuthenticationToken(TestConstants.TESTUSER, TestConstants.TESTPASSWORD, AuthorityUtils.createAuthorityList("ROLE_DUMMY")));
+
+        Assertions.assertThatExceptionOfType(AccessDeniedException.class)
+                .isThrownBy(() -> broadcastService.getOldestMessage(null))
+                .withMessageStartingWith("Access Denied");
+    }
+
+    @Test
+    void deleteAccessDeniedTest() {
+        SecurityContextHolder.getContext().setAuthentication(
+                new UsernamePasswordAuthenticationToken(TestConstants.TESTUSER, TestConstants.TESTPASSWORD, AuthorityUtils.createAuthorityList("ROLE_DUMMY")));
+
+        Assertions.assertThatExceptionOfType(AccessDeniedException.class)
+                .isThrownBy(() -> broadcastService.deleteMessage(null))
+                .withMessageStartingWith("Access Denied");
+    }
+
+    @Test
+    void broadcastAccessPositiveTest() {
```

Comment:
der Test heißt `accessPositiv` daher würde ich davon ausgehen es werden Szenarien geprüft die durchgehen durch die Sec. Enthalten sind aber auch Fälle wo das korrekten Werfen einer Exception verifiziert wird.

Es werden mehrere Testfälle gemischt. In einer Testmethode sollte immer nur ein Szenario geprüft werden.